extensions/typescript/src/utils/telemetry.ts

```diff
@@ -23,8 +24,16 @@ export default class TelemetryReporter {
 		}
 	}
 
+	constructor(private client: TypeScriptServiceClient) {
```

Comment:
This creates a circular dependency between the reporter and the service client. Can we either pass the version info in `logTelemetry` or make `TelemetryReporter` take an interface instead of the entire client

Reply:
Exposed the client version through a new `IClientVersion` interface placed in `typescriptServiceClient.ts`. So a circular dependency remains. This is preferable to passing the version explicitly because other services may need to report telemetry in the future, and it would be inconvenient to force them to have a handle on the tsserver version.